pkg/operator/starter.go

```diff
@@ -350,3 +324,61 @@ var CertSecrets = []installer.UnrevisionedResource{
 	// and in the PDB
 	{Name: "etcd-all-certs"},
 }
+
+// ensureLegacyPDBCleanup continually ensures the removal of the legacy v1beta1 PDB resource.
+// TODO remove in 4.10
+func ensureLegacyPDBCleanup(ctx context.Context, kubeClient *kubernetes.Clientset, infrastructureLister configv1listers.InfrastructureLister, eventRecorder events.Recorder) {
+	resourceName := "etcd-quorum-guard"
+	legacyPDBClient := kubeClient.PolicyV1beta1().PodDisruptionBudgets(operatorclient.TargetNamespace)
+	etcdOClient := kubeClient.AppsV1().Deployments(operatorclient.TargetNamespace)
+
+	go wait.UntilWithContext(ctx, func(_ context.Context) {
+		// This function isn't expected to take long enough to suggest
+		// checking that the context is done. The wait method will do that
+		// checking.
+
+		controlPlaneTopology, err := ceohelpers.GetControlPlaneTopology(infrastructureLister)
+		if err != nil {
+			klog.Warningf("Error retrieving control-plane topology: %w", err)
+			return
+		}
+		if controlPlaneTopology == configv1.SingleReplicaTopologyMode {
+			// Done SNO does not deploy PDB
+			return
+		}
+
+		// Check whether the legacy pod disruption budget exists and is not marked for deletion
+		legacyPDB, err := legacyPDBClient.Get(ctx, resourceName, metav1.GetOptions{})
+		if errors.IsNotFound(err) {
+			// Done - legacy PDB does not exist
+			return
+		}
+		if err != nil {
+			klog.Warningf("Error retrieving legacy pod disruption budget: %v", err)
+			return
+		}
+		if legacyPDB.ObjectMeta.DeletionTimestamp != nil {
+			// Done - resource has been marked for deletion
+			return
+		}
+
+		// Check that the deployment managing the apiserver pods has the correct number of replicas
+		etcdOperatorQuorumGuard, err := etcdOClient.Get(ctx, resourceName, metav1.GetOptions{})
+		if errors.IsNotFound(err) {
+			// No available replicas if the deployment doesn't exist
+			return
+		}
+		if etcdOperatorQuorumGuard.Status.AvailableReplicas == etcdOperatorQuorumGuard.Status.Replicas {
```

Comment:
Maybe comment or make it clear in the code why `AvailableReplicas == Replicas` is indicative of not having all available replicas?

Reply:
thanks clearly wrong operator 0:)